### PR TITLE
Move `leaf` and `router` role from metal-roles into mini-lab.

### LINKED ISSUE #155: Move `leaf` and `router` role from metal-roles into mini-lab
These roles were placed into metal-roles but they are only used in the mini-lab.

### PR DESCRIPTION
**Pull request by Gerrit91**
Closes #155.

### CODE REVIEW

**Comment by robertvolkmann** [comment]
Maybe rename the role `leaf` to `cumulus`.

File: roles/cumulus/tasks/router.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+- name: render frr configuration
+  template:
+    src: frr.conf.j2
+    dest: /etc/frr/frr.conf
+    validate: '/usr/bin/vtysh --dryrun --inputfile %s'
+  tags: frr
+  register: frr_rendered
+  notify: reload frr
+  when: "ansible_facts.services['metal-core.service'] is not defined"
```
Comment (robertvolkmann) [inline roles/cumulus/tasks/router.yaml:79]: Not needed for a leaf

File: roles/cumulus/tasks/router.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+- name: populate service facts
+  service_facts:
+
+- name: render interfaces configuration
+  template:
+    src: interfaces.j2
+    dest: /etc/network/interfaces
+    validate: '/sbin/ifup --syntax-check --all --interfaces %s'
+  notify: reload interfaces
+  when: "ansible_facts.services['metal-core.service'] is not defined"
```
Comment (robertvolkmann) [inline roles/cumulus/tasks/router.yaml:49]: Not needed for a leaf

**Comment by Gerrit91** [comment]
Fails when I apply your requested changes:

```
deploy-partition      | TASK [cumulus : ensure that static route for return path to pxe network is present] ***
deploy-partition      | fatal: [leaf02]: FAILED! => changed=true 
deploy-partition      |   cmd:
deploy-partition      |   - ip
deploy-partition      |   - r
deploy-partition      |   - a
deploy-partition      |   - 10.0.1.0/24
deploy-partition      |   - vrf
deploy-partition      |   - mgmt
deploy-partition      |   - via
deploy-partition      |   - 10.0.1.1
deploy-partition      |   - dev
deploy-partition      |   - vlan4000
deploy-partition      |   delta: '0:00:00.063829'
deploy-partition      |   end: '2024-05-03 06:25:57.705250'
deploy-partition      |   msg: non-zero return code
deploy-partition      |   rc: 1
deploy-partition      |   start: '2024-05-03 06:25:57.641421'
deploy-partition      |   stderr: Cannot find device "vlan4000"
deploy-partition      |   stderr_lines: <omitted>
deploy-partition      |   stdout: ''
deploy-partition      |   stdout_lines: <omitted>
deploy-partition      | fatal: [leaf01]: FAILED! => changed=true 
deploy-partition      |   cmd:
deploy-partition      |   - ip
deploy-partition      |   - r
deploy-partition      |   - a
deploy-partition      |   - 10.0.1.0/24
deploy-partition      |   - vrf
deploy-partition      |   - mgmt
deploy-partition      |   - via
deploy-partition      |   - 10.0.1.1
deploy-partition      |   - dev
deploy-partition      |   - vlan4000
deploy-partition      |   delta: '0:00:00.071334'
deploy-partition      |   end: '2024-05-03 06:25:57.692110'
deploy-partition      |   msg: non-zero return code
deploy-partition      |   rc: 1
deploy-partition      |   start: '2024-05-03 06:25:57.620776'
deploy-partition      |   stderr: Cannot find device "vlan4000"
deploy-partition      |   stderr_lines: <omitted>
deploy-partition      |   stdout: ''
deploy-partition      |   stdout_lines: <omitted>
```

Please feel free to commit to this pull request directly if you want changes for the switch configuration as I will not be able to judge those. I only moved existing tasks with this PR.

**Review by robertvolkmann — APPROVED** [review]
Looks fine. Reworking will be done in the next PR.

**Pull request merged**